source/tutorials/contributing-to-nixpkgs.md

```diff
@@ -0,0 +1,204 @@
+
+```nix
+# icat.nix
+...
+  meta = with lib; {
+    description = "icat (Image cat) outputs images in 256-color capable terminals.";
+    homepage = "https://github.com/atextor/icat";
+    license = licenses.bsdOriginal;
+    platforms = platforms.unix;
+  };
+...
+```
+
+With no further technical changes necessary, this derivation is now complete.
+
+## Adding to Nixpkgs
+You can now clone a local copy of `nixpkgs`, and move your derivation into an appropriate location within that source tree.
+
+```console
+$ git clone https://github.com/nixos/nixpkgs
+```
+
+`icat` is a console tool which also displays graphics, so there are a few places within Nixpkgs that you could reasonably move it.
+
+There are currently two other similar tools already in `nixpkgs`, both of which are stored in `pkgs/applications/graphics`, so this is a good candidate for a home for `icat` too.
+
+```console
+$ mkdir nixpkgs/pkgs/applications/graphics/icat
+```
+
+You should rename `icat.nix` when you copy it to its new home, to keep to the convention of package derivation files being named `default.nix`:
+
+```console
+$ mv icat.nix nixpkgs/pkgs/applications/graphics/icat/default.nix
+```
+
+The relative directory paths to all packages get referenced in `pkgs/top-level/all-packages.nix`, which you should update now, putting the new package next to the other similar tools:
+
+```diff
+# all-packages.nix
+...
++ icat = callPackage ../applications/graphics/icat { };
+
+  imgcat = callPackage ../applications/graphics/imgcat { };
+
+  img-cat = callPackage ../applications/graphics/img-cat { };
+...
+```
+
+:::{note}
+Attributes in `all-packages.nix` are arranged *approximately* in alphabetical order, either by attribute name or by the directory containing their corresponding `default.nix`.
+:::
+
+## One Final Test
+To confirm that the package builds within its new context, you should change directories to the root of the Nixpkgs repo and run `nix-build -A icat` a final time:
+
+```console
+$ cd nixpkgs
+$ nix-build -A icat
+this derivation will be built:
+  /nix/store/kf7na78dhwhfrgxjaj27gjkqs4b3n7wz-icat.drv
+...
+/nix/store/vr2vk8z8839l5j6gra0qlyxrh5sarmh4-icat
+```
+
+The build completed as expected, and produced a `result` in the current directory.
+
+The package is now ready for submission to Nixpkgs!
+
+## Next Steps
+With the package building successfully, you're almost ready to contribute your work to Nixpkgs.
+
+Be sure to read the [contributing documentation](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md) to understand the current submission conventions before proceeding.
+
+:::{note}
+Before contributing your first package, you *must* add your information to [`nixpkgs/maintainers/maintainers-list.nix`](https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix), following the instructions [here](https://nixos.org/manual/nixpkgs/stable/#var-meta-maintainers).
```

Comment:
This is a better docs link: https://github.com/NixOS/nixpkgs/tree/master/maintainers#readme